invenio/modules/accounts/models.py

```diff
@@ -85,12 +90,74 @@ def password(self):
     @password.setter
     def password(self, password):
         """Set the password."""
-        self._password = self._password_comparator.hash(password)
+        if password is None:
+            # Unusable password.
+            self._password = None
+            self.password_scheme = ''
+        else:
+            self._password = password_context.encrypt(password)
+            self.password_scheme = password_context.default_scheme()
+
+        # Invenio legacy salt is stored in password_salt, and every new
+        # password set will be migrated to new hash not relying on
+        # password_salt, thus is force to empty value.
+        self.password_salt = ""
+
+    def verify_password(self, password, migrate=False):
+        """Verify if password matches the stored password hash."""
+        if self.password is None or password is None:
+            return False
+
+        # Invenio 1.x legacy needs externally store password salt to compute
+        # hash.
+        scheme_ctx = {} if \
+            self.password_scheme != invenio_aes_encrypted_email.name else \
+            {'user': self.password_salt}
+
+        # Verify password
+        if not password_context.verify(password, self.password,
+                                       scheme=self.password_scheme,
+                                       **scheme_ctx):
+                return False
+
+        # Migrate hash if needed.
+        if migrate and password_context.needs_update(self.password):
+            self.password = password
+            db.session.commit()
 
-    @password.comparator
-    def password(self):
-        """Compare password."""
-        return self._password_comparator
+        return True
+
+    def verify_email(self, force=False):
+        """Verify email address."""
+        if force or self.note == "2":
```

Comment:
@PXke you can happily ignore this line as the `User.note` is going to be splitted to several columns like: email confirmed, user activated, etc.